cvat/apps/quality_control/models.py

```diff
@@ -76,26 +77,38 @@ class QualityReport(models.Model):
     task = models.ForeignKey(
         Task, on_delete=models.CASCADE, related_name="quality_reports", null=True, blank=True
     )
-
-    parent = models.ForeignKey(
-        "self", on_delete=models.CASCADE, related_name="children", null=True, blank=True
+    project = models.ForeignKey(
+        Project, on_delete=models.CASCADE, related_name="quality_reports", null=True, blank=True
     )
+
+    # job reports should all have a single parent report
+    # task reports may have none, or be shared between several project reports
```

Comment:
It could also be implemented for jobs, but I didn't do this to simplify logic. In the case of projects, a task can be unchanged from the previous project report computation. In this case, the previous task report is used in the new project report.